pkg/types/automation/interfaces.go

```diff
@@ -0,0 +1,139 @@
+package automation
+
+import (
+	"context"
+	"io"
+)
+
+type UpkeepTypeGetter func(UpkeepIdentifier) UpkeepType
+type WorkIDGenerator func(UpkeepIdentifier, Trigger) string
+
+// UpkeepStateStore is the interface for managing upkeeps final state in a local store.
+type UpkeepStateStore interface {
+	UpkeepStateUpdater
+	UpkeepStateReader
+	Start(context.Context) error
+	io.Closer
+}
+
+type Registry interface {
+	CheckUpkeeps(ctx context.Context, keys ...UpkeepPayload) ([]CheckResult, error)
+	Name() string
+	Start(ctx context.Context) error
+	Close() error
+	HealthReport() map[string]error
+}
+
+type EventProvider interface {
+	Name() string
+	Start(_ context.Context) error
+	Close() error
+	Ready() error
+	HealthReport() map[string]error
```

Comment:
You can use `services.Service` for this common set.
```suggestion
	services.Service
```
You might want/need to upgrade anything with Start/Close to `services.Service` also, since that suggests they have background processes/state that could be monitored.

Reply:
Otherwise there is `services.StartClose`:
```go
type StartClose interface {
	Start(context.Context) error
	Close() error
}
```